pandas/core/reshape/reshape.py

```diff
@@ -476,12 +499,13 @@ def _unstack_frame(obj, level, fill_value=None):
         unstacker = partial(_Unstacker, index=obj.index,
                             level=level, fill_value=fill_value)
         blocks = obj._data.unstack(unstacker)
-        klass = type(obj)
+        klass = obj._constructor
         return klass(blocks)
     else:
         unstacker = _Unstacker(obj.values, obj.index, level=level,
```

Comment:
might be a slightly bigger change, but could pass ``obj`` to ``_Unstacker`` directly and *might* be simpler.